mirascope/chat/parsers.py

```diff
@@ -0,0 +1,68 @@
+"""Classes for using parsers with Chat APIs."""
+import json
+from typing import Callable, Generator, Optional, Type, Union
+
+from openai.types.chat.chat_completion_message_tool_call import (
+    ChatCompletionMessageToolCall,
+    Function,
+)
+from pydantic import BaseModel
+
+from ..partial import Partial
+from .tools import OpenAITool
+from .types import OpenAIChatCompletionChunk
+from .utils import convert_tools_list_to_openai_tools
+
+
+class PartialOpenAIToolParser(BaseModel):
```

Comment:
re: discussion, let's rename this to something like `OpenAIToolStreamParser` and clean up the below functionality such that it will always return a full tool (and not partial, so the consumer end is cleaner cause they can always expect either a full tool or a validation error)